src/org/labkey/test/components/react/dialogs/BulkCreateSamplesDialog.java

```diff
@@ -0,0 +1,159 @@
+package org.labkey.test.components.react.dialogs;
+
+import org.labkey.test.BootstrapLocators;
+import org.labkey.test.Locator;
+import org.labkey.test.components.bootstrap.ModalDialog;
+import org.labkey.test.components.glassLibrary.components.FilteringReactSelect;
+import org.labkey.test.components.glassLibrary.components.ReactSelect;
+import org.openqa.selenium.StaleElementReferenceException;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+
+import java.util.List;
+
+public class BulkCreateSamplesDialog extends ModalDialog
+{
+
+    public BulkCreateSamplesDialog(WebDriver driver)
+    {
+        this(new ModalDialogFinder(driver).withTitle("Bulk Creation of Samples"));
+    }
+
+    private BulkCreateSamplesDialog(ModalDialogFinder finder)
+    {
+        super(finder);
+    }
+
+    public BulkCreateSamplesDialog setQuantity(int quantity)
+    {
+        return setQuantity(Integer.toString(quantity));
+    }
+
+    public BulkCreateSamplesDialog setQuantity(String quantity)
+    {
+        getWrapper().setFormElement(elementCache().quantity, quantity);
+        return this;
+    }
+
+    public String getQuantity()
+    {
+        return getWrapper().getFormElement(elementCache().quantity);
+    }
+
+    public BulkCreateSamplesDialog setDescription(String description)
+    {
+        getWrapper().setFormElement(elementCache().description, description);
+        return this;
+    }
+
+    public String getDescription()
+    {
+        return getWrapper().getFormElement(elementCache().description);
+    }
+
+    public BulkCreateSamplesDialog setTextField(String fieldCaption, String value)
+    {
+        String forField = Locator.xpath("//div[@class='modal-body']//label[./span[contains(text(), '" + fieldCaption + "')]]").findElement(getComponentElement()).getAttribute("for");
+        getWrapper().setFormElement(Locator.tagWithId("input", forField), value);
+        return this;
+    }
+
+    public String getTextField(String fieldCaption)
+    {
+        String forField = Locator.xpath("//div[@class='modal-body']//label[./span[contains(text(), '" + fieldCaption + "')]]").findElement(getComponentElement()).getAttribute("for");
+        return getWrapper().getFormElement(Locator.tagWithId("input", forField));
+    }
+
+    private boolean waitForReactSelectReady(ReactSelect select)
+    {
+        getWrapper().waitFor(()->!select.getSelections().contains("Loading..."), "The reactSelect field did not populate in time.", 1_000);
+        return true;
+    }
+
+    public BulkCreateSamplesDialog setSelectionField(String fieldCaption, List<String> selectValues)
+    {
+        FilteringReactSelect reactSelect = FilteringReactSelect.finder(getDriver()).followingLabelWithSpan(fieldCaption).find();
+        getWrapper().waitFor(()-> reactSelect.isInteractive(), 5000);
```

Comment:
Are you sure this will work?
My experience was that the arrow on the reactSelect was showing up enabled when data was still loading. That is why the waitForReactSelectReady was added.

Reply:
I'm willing to be wrong on this.  But, if waitForReactSelectReady works, I'd be happy to try to replace the implementation of isInteractive() with what works there

Reply:
Ok, that sounds like a good approach. The call to .isInteractive() is not a reliable check if the dropdown control is a look up field. This is why the [SMSampleCreateTest.testBulkInsert](https://teamcity.labkey.org/buildConfiguration/LabKey_Trunk_Premium_SampleManager_SampleManagerAPostgres/1055312?buildTab=tests&pager.currentPage=1) is failing in the branch. A call to isInteractive is returning true but there is a split second between that and the list int he drop down being populated, which is enough time for the test code to type the text which is not found.

Reply:
At the moment, SMSampleCreateTest.testBulkInsert is failing for a different reason.
By switching to use FilteringReactSelect.filterSelect(), the test filters the input, and looks for a value that contains the filter clause- but values here don't contain the full filter being supplied.  Easy workaround: omit the Color: prefix, just search for the color ("orange")
I've sent you a patch with this workaround, wanted to give you the input on whether it's the right fix